plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,834 @@
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_interface_setting
+short_description: Manage Interface Policy Groups on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage Interface Policy Groups on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Anvitha Jain (@anvjain)
+options:
+  template:
+    description:
+    - The name of the template.
+    - The template must be a fabric policy template.
+    type: str
+    required: true
+  interface_policy_group:
+    description:
+    - The name of the interface policy group.
+    type: str
+    aliases: [ name ]
+  interface_policy_group_uuid:
+    description:
+    - The UUID of the interface policy group.
+    - This parameter is required when the O(interface_policy_group) needs to be updated.
+    type: str
+    aliases: [ uuid ]
+  description:
+    description:
+    - The description of the interface policy group.
+    type: str
+  interface_type:
+    description:
+    - The type of the interface policy group.
+    - The default value is C(physical).
+    type: str
+    choices: [ physical, port_channel ]
+  speed:
+    description:
+    - The data transfer rate for the port in interface policy group.
+    - The default value is C(inherit).
+    type: str
+    choices: [ 100M, 1G, 10G, 25G, 40G, 50G, 100G, 200G, 400G, inherit ]
+  auto_negotiation:
+    description:
+    - The auto negotiation of the port in interface policy group.
+    - The default value is C(on).
+    type: str
+    choices: [ 'on', 'off', on_enforce ]
+  vlan_scope:
+    description:
+    - The VLAN encapsulation value to map to EPG.
+    - The default value is C(global).
+    type: str
+    choices: [ global, port_local ]
+  cdp_admin_state:
+    description:
+    - The CDP admin state enables Cisco Discovery Protocol (CDP) on the interface.
+    - The default value is C(disabled).
+    type: str
+    choices: [ enabled, disabled ]
+  domains:
+    description:
+    - The domains with which you want to associate this interface policy.
+    - The domains must be defined in the same fabric policy template.
+    - The old O(domains) will be replaced by the new entries during an update.
+    - Providing an empty list will remove the O(domains) from the interface policy.
+    type: list
+    elements: str
+  port_channel_mode:
+    description:
+    - The port channel mode of the interface policy group.
+    - The default value is C(static_channel_mode_on).
+    - The value is available only when the interface_type is C(port_channel).
+    type: str
+    choices: [ static_channel_mode_on, lacp_passive, lacp_active,  mac_pinning, mac_pinning_physical_nic_load, use_explicit_failover_order ]
```

Comment:
extra white space between `lacp_active` and `mac_pinning`

Reply:
removed